tests/telemetry/conftest.py

```diff
@@ -106,3 +106,19 @@ def test_agent_session(telemetry_writer, request):
         telemetry_writer.periodic()
         del telemetry_writer._client._headers["X-Datadog-Test-Session-Token"]
         del os.environ["_DD_TELEMETRY_WRITER_ADDITIONAL_HEADERS"]
+
+
+@pytest.fixture
+def test_agent_session_telemetry_metrics(test_agent_session):
+    test_agent_session.telemetry_writer._flush_namespace_metrics()
+    test_agent_session.telemetry_writer._flush_events_queue()
+    test_agent_session.telemetry_writer._flush_integrations_queue()
```

Comment:
We should update telemetry_writer._reset() to flush metrics queues

```suggestion
    test_agent_session.telemetry_writer._reset()
```

Reply:
Updated with the last refactor, thanks for the suggestion!